File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int ans = search(nums, target);
        return ans;
    }
    
    int search(int[] arr, int target){
        int start = 0;
        int end = arr.length-1;
        
        while(start<=end){
            int mid = start + (end - start)/2;
            
            if(arr[mid]==target){
                return mid;
            }
            
            if(arr[mid]>target){
                end = mid-1;
            }else{
                start = mid+1;
            }
        }
        return start;
    }
}

File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int search(vector<int> &arr, int target){
    int n = arr.size();
    int start = 0, end = n - 1;
    while(start<=end){
        int mid = (start + end) / 2;
        if(arr[mid] == target) return mid;

        if(arr[mid]>target){
            end = mid-1;
        }else{
            start = mid+1;
        }

    }
        return start;
}
    int searchInsert(vector<int>& nums, int target) {
        return search(nums, target);
    }
};